python/lsst/ctrl/mpexec/singleQuantumExecutor.py

```diff
@@ -476,8 +477,39 @@ def runQuantum(
             task.runQuantum(butlerQC, inputRefs, outputRefs)
         except NoWorkFound as err:
             # Not an error, just an early exit.
-            _LOG.info("Task '%s' on quantum %s exited early: %s", task_node.label, quantum.dataId, str(err))
-            pass
+            _LOG.info(
+                "Task '%s' on quantum %s exited early with no work found: %s.",
+                task_node.label,
+                quantum.dataId,
+                str(err),
+            )
+        except AnnotatedPartialOutputsError as caught:
+            error: BaseException
+            if caught.__cause__ is None:
+                _LOG.warning(
+                    "Incorrect use of AnnotatedPartialOutputsError: no chained exception found.",
+                    task_node.label,
+                    quantum.dataId,
+                )
+                error = caught
+            else:
+                error = caught.__cause__
+            if self.raise_on_partial_outputs:
+                # This is a real edge case that required some experimentation:
```

Comment:
I think I'd put a `NOTE:` or something at the start of this, as it's sort of a meta-comment.